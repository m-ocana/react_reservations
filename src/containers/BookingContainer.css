
.booking_container {
  display: flex;
}

.booking_entries {
  flex: 2;
  overflow: auto;
}

.date {
  font-size: 1rem;
}
