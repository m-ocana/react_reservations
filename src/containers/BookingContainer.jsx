/* Feel free to edit */
import React, { Component } from 'react'

// If you use React Router, make this component
// render <Router> with your routes. Currently,
// only synchronous routes are hot reloaded, and
// you will see a warning from <Router> on every reload.
// You can ignore this warning. For details, see:
// https://github.com/reactjs/react-router/issues/2182
export default class BookingContainer extends Component {
  constructor(props) {
    super(props)
    this.state = { bookingEntries: [] }
  }

  componentDidMount() {
    fetch('http://localhost:3000/bookings.json')
      .then((response) => {
        return response.json()
      })
      .then((bookingEntries) => {
        this.setState({ bookingEntries })
      })
  }

  renderBookingEntries = () => {
    return null
  }

  render() {
    const { bookingEntries } = this.state
    return (
      <div>
        { bookingEntries.length ? this.renderBookingEntries()
          : <span>Sorry, no bookings were found</span>
        }
      </div>
    )
  }
}
