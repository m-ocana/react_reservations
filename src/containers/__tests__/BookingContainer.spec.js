import toJson from 'enzyme-to-json'
import renderComponent from '../../helpers/test-component'
import BookingContainer from '../BookingContainer'

const mockedState = {
  bookingEntries: [{
    date: '2017-03-13',
    bookings: [
      {
        title: 'Mr',
        firstName: 'Fred',
        lastName: 'Bloggs',
        time: '16.00',
        partySize: 2,
        seated: false,
        cancelled: false,
        notes: ''
      }
    ]
  }]
}
const renderContainer = renderComponent(BookingContainer)

describe('<BookingContainer /> ', () => {
  describe('@renders', () => {
    it('should render BookingContainer without bookings', () => {
      expect(renderContainer().getTree())
        .toMatchSnapshot()
    })

    it('should render BookingContainer with bookings', () => {
      const { wrapper } = renderContainer()
      wrapper.setState({ mockedState })
      expect(toJson(wrapper))
        .toMatchSnapshot()
    })
  })

  describe('@lifecycle', () => {
    describe('componentDidMount', () => {
      it('should fetch data', () => {
        window.fetch = jest.fn().mockImplementation(() =>
          Promise.resolve())

        const { instance } = renderContainer()
        expect(window.fetch).not.toHaveBeenCalled()
        instance.componentDidMount()
        expect(window.fetch).toHaveBeenCalled()
      })
    })
  })
})
