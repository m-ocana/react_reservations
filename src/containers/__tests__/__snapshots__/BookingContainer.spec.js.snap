// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BookingContainer />  @renders should render BookingContainer with bookings 1`] = `
<div>
  <span>
    Sorry, no bookings were found
  </span>
</div>
`;

exports[`<BookingContainer />  @renders should render BookingContainer without bookings 1`] = `
<div>
  <span>
    Sorry, no bookings were found
  </span>
</div>
`;
